pyuvdata/uvdata/uvfits.py

```diff
@@ -1000,8 +1009,20 @@ def write_uvfits(
             time_array1 = self.time_array - jd_midnight
         int_time_array = self.integration_time
 
+        # If using MIRIAD convention, we need 1-indexed data
+        if use_miriad_convention:
+            if np.min(self.antenna_numbers) == 0:
+                self.antenna_numbers += 1
+                self.ant_1_array += 1
+                self.ant_2_array += 1
```

Comment:
If you change these values on the object, you really need to update the baseline array on the object so that it's not wrong. I'd actually prefer to not update these on the object at all, but just to use these updates in the file itself.